From f6886c2c33a1cf8771163919f3d20f6340c0ce38 Mon Sep 17 00:00:00 2001
From: Eelco Dolstra <eelco.dolstra@logicblox.com>
Date: Fri, 10 Jul 2015 18:12:37 +0200
Subject: [PATCH] Quick fix for #47

https://github.com/NixOS/patchelf/issues/47

Avoid issues with holes in binaries such as qemu-pcc from qemu-native.

Upstream-Status: Submitted
RP
2016/2/3

---
 src/patchelf.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: patchelf/src/patchelf.cc
===================================================================
--- patchelf.orig/src/patchelf.cc	2016-08-11 15:51:28.142895785 -0600
+++ patchelf/src/patchelf.cc	2016-08-11 15:51:49.006896065 -0600
@@ -256,7 +256,7 @@
     struct stat st;
     if (stat(fileName.c_str(), &st) != 0) error("stat");
     fileSize = st.st_size;
-    maxSize = fileSize + 32 * 1024 * 1024;
+    maxSize = fileSize + 64 * 1024 * 1024;
 
     contents = (unsigned char *) malloc(fileSize + maxSize);
     if (!contents) abort();
